src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileMetadataService.scala

```diff
@@ -83,43 +81,80 @@ class FileMetadataService(
   }
 
   def deleteFileMetadata(input: DeleteFileMetadataInput, userId: UUID): Future[DeleteFileMetadata] = {
-
+    val customMetadataService = new CustomMetadataPropertiesService(customMetadataPropertiesRepository)
     for {
       existingFileRows <- fileRepository.getAllDescendants(input.fileIds.distinct)
       fileIds: Set[UUID] = existingFileRows.toFileTypeIds
-      metadataValues <- customMetadataPropertiesRepository.getCustomMetadataValues
-      groupId = metadataValues
-        .find(property => property.propertyname == ClosureType && property.propertyvalue == "Closed")
-        .map(_.dependencies)
-        .getOrElse(throw new IllegalStateException("Can't find metadata property 'ClosureType' with value 'Closed' in the db"))
-      dependencies <- customMetadataPropertiesRepository.getCustomMetadataDependencies
-      propertyNames = dependencies.filter(dependency => groupId.contains(dependency.groupid)).map(_.propertyname)
-
-      (fileMetadataToUpdate, fileMetadataToDelete) = getFileMetadataToUpdateAndDelete(metadataValues, propertyNames, userId)
-      _ <- fileMetadataRepository.updateFileMetadataProperties(fileIds, fileMetadataToUpdate)
-      _ <- fileMetadataRepository.deleteFileMetadata(fileIds, fileMetadataToDelete.toSet)
-    } yield DeleteFileMetadata(fileIds.toSeq, propertyNames)
-  }
+      customMetadataProperties <- customMetadataService.getCustomMetadata
+      namesOfAllPropertiesToDelete: Set[String] = customMetadataProperties
+        .collect {
+          case customMetadataProperty if input.propertyNames.contains(customMetadataProperty.name) =>
+            val namesOfDependenciesToDelete: List[String] = customMetadataProperty.values.flatMap(_.dependencies.map(_.name))
+            namesOfDependenciesToDelete :+ customMetadataProperty.name
+        }
+        .flatten
+        .toSet
+
+      _ = if (namesOfAllPropertiesToDelete.isEmpty) {
```

Comment:
Don't think there is any need to throw an exception, if there is nothing to delete there is nothing to delete.